az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -352,24 +365,50 @@ public static void proxyUserKillAllSpawnedHadoopJobs(
     final Properties properties = new Properties();
     properties.putAll(jobProps.getFlattened());
 
-    try {
-      if (HadoopSecureWrapperUtils.shouldProxy(properties)) {
-        final UserGroupInformation proxyUser =
-            HadoopSecureWrapperUtils.setupProxyUserWithHSM(hadoopSecurityManager, properties,
-                tokenFile.getAbsolutePath(), log);
-        proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
-          @Override
-          public Void run() throws Exception {
-            findAndKillYarnApps(jobProps, log);
-            return null;
-          }
-        });
-      } else {
-        findAndKillYarnApps(jobProps, log);
+    // todo: use feature flag
+    String yarnKillVersion = jobProps.getString(YARN_KILL_VERSION, YARN_KILL_LEGACY);
+    if (YARN_KILL_LEGACY.equals(yarnKillVersion)) {
```

Comment:
How about leaving this to "else" instead of using "if & elseif"?
I'm afraid yarnKillVersion != YARN_KILL_LEGACY && yarnKillVersion != YARN_KILL_TOKEN_FIX by manual error.

Reply:
Agreed. The behavior should be "default is legacy". So something like,

if (YARN_KILL_TOKEN_FIX) { 
  new logic;
} else {
  old logic;
}